integration-tests/test/shared/env.ts

```diff
@@ -65,82 +45,34 @@ export class OptimismEnv {
   verifierProvider: providers.JsonRpcProvider
 
   constructor(args: any) {
-    this.addressManager = args.addressManager
-    this.l1Bridge = args.l1Bridge
-    this.l1Messenger = args.l1Messenger
-    this.l1BlockNumber = args.l1BlockNumber
-    this.ovmEth = args.ovmEth
-    this.l2Bridge = args.l2Bridge
-    this.l2Messenger = args.l2Messenger
-    this.gasPriceOracle = args.gasPriceOracle
-    this.sequencerFeeVault = args.sequencerFeeVault
     this.l1Wallet = args.l1Wallet
     this.l2Wallet = args.l2Wallet
     this.messenger = args.messenger
     this.l1Provider = args.l1Provider
     this.l2Provider = args.l2Provider
     this.replicaProvider = args.replicaProvider
     this.verifierProvider = args.verifierProvider
-    this.ctc = args.ctc
-    this.scc = args.scc
   }
 
   static async new(): Promise<OptimismEnv> {
     const network = await l1Provider.getNetwork()
 
-    const addressManager = getAddressManager(l1Wallet)
-    const l1Bridge = await getL1Bridge(l1Wallet, addressManager)
-
-    const l1MessengerAddress = await addressManager.getAddress(
-      'Proxy__OVM_L1CrossDomainMessenger'
-    )
-    const l2MessengerAddress = await addressManager.getAddress(
-      'L2CrossDomainMessenger'
-    )
-    const l1Messenger = getContractFactory('L1CrossDomainMessenger')
-      .connect(l1Wallet)
-      .attach(l1MessengerAddress)
-    const ovmEth = getOvmEth(l2Wallet)
-    const l2Bridge = await getL2Bridge(l2Wallet)
-    const l2Messenger = getContractFactory('L2CrossDomainMessenger')
-      .connect(l2Wallet)
-      .attach(l2MessengerAddress)
-    const ctcAddress = await addressManager.getAddress(
-      'CanonicalTransactionChain'
-    )
-    const ctc = getContractFactory('CanonicalTransactionChain')
-      .connect(l1Wallet)
-      .attach(ctcAddress)
-
-    const gasPriceOracle = getContractFactory('OVM_GasPriceOracle')
-      .connect(gasPriceOracleWallet)
-      .attach(predeploys.OVM_GasPriceOracle)
-
-    const sccAddress = await addressManager.getAddress('StateCommitmentChain')
-    const scc = getContractFactory('StateCommitmentChain')
-      .connect(l1Wallet)
-      .attach(sccAddress)
-
-    const sequencerFeeVault = getContractFactory('OVM_SequencerFeeVault')
-      .connect(l2Wallet)
-      .attach(predeploys.OVM_SequencerFeeVault)
-
-    const l1BlockNumber = getContractFactory('iOVM_L1BlockNumber')
-      .connect(l2Wallet)
-      .attach(predeploys.OVM_L1BlockNumber)
-
     const messenger = new CrossChainMessenger({
       l1SignerOrProvider: l1Wallet,
       l2SignerOrProvider: l2Wallet,
       l1ChainId: network.chainId,
       contracts: {
         l1: {
```

Comment:
You can actually leave this out, the SDK will pick the contract addresses automatically based on the chain ID.

Reply:
cc @mslipper I just need the contract addresses for nightly so that it will work correctly in the integration tests.

Reply:
@tonykogias let's remove the entire `contracts` object here, should be pulled in automatically

Reply:
> cc @mslipper I just need the contract addresses for nightly so that it will work correctly in the integration tests.

We should be able to support addresses that aren't hardcoded - the point of nightly is that it can be blown away and fully redeployed at any moment